bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -0,0 +1,448 @@
+                    .withLabel(FAN_LEVEL_LABEL).withType(fanLevelChannelType).build());
+        });
+
+        final ChannelTypeUID modeChannelType = addChannelType(SensiboBindingConstants.CHANNEL_TYPE_MODE, MODE_LABEL,
+                "String", sensiboSky.getRemoteCapabilities().keySet(), null, null);
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_MODE), "String")
+                .withLabel(MODE_LABEL).withType(modeChannelType).build());
+
+        final ChannelTypeUID targetTemperatureChannelType = addChannelType(
+                SensiboBindingConstants.CHANNEL_TYPE_TARGET_TEMPERATURE, TARGET_TEMPERATURE_LABEL, "Number:Temperature",
+                sensiboSky.getTargetTemperatures(), "%d %unit%", "TargetTemperature");
+        newChannels.add(ChannelBuilder
+                .create(new ChannelUID(getThing().getUID(), SensiboBindingConstants.CHANNEL_TARGET_TEMPERATURE),
+                        "Number:Temperature")
+                .withLabel(TARGET_TEMPERATURE_LABEL).withType(targetTemperatureChannelType).build());
+
+        return newChannels;
+    }
+
+    private ChannelTypeUID addChannelType(final String channelTypePrefix, final String label, final String itemType,
+            final Collection<?> options, @Nullable final String pattern, @Nullable final String tag) {
+        final ChannelTypeUID channelTypeUID = new ChannelTypeUID(SensiboBindingConstants.BINDING_ID,
+                channelTypePrefix + getThing().getUID().getId());
+        final List<StateOption> stateOptions = options.stream()
+                .map(e -> new StateOption(e.toString(), e instanceof String ? beautify((String) e) : e.toString()))
+                .collect(Collectors.toList());
+
+        StateDescriptionFragmentBuilder stateDescription = StateDescriptionFragmentBuilder.create().withReadOnly(false)
+                .withOptions(stateOptions);
+        if (pattern != null) {
+            stateDescription = stateDescription.withPattern(pattern);
+        }
+        final StateChannelTypeBuilder builder = ChannelTypeBuilder.state(channelTypeUID, label, itemType)
+                .withStateDescription(stateDescription.build().toStateDescription());
+        if (tag != null) {
+            builder.withTag(tag);
+        }
+        final ChannelType channelType = builder.build();
+
+        generatedChannelTypes.put(channelTypeUID, channelType);
+
+        return channelTypeUID;
+    }
+
+    @Override
+    public Collection<ChannelType> getChannelTypes(@Nullable final Locale locale) {
+        return generatedChannelTypes.values();
+    }
+
+    @Override
+    public @Nullable ChannelType getChannelType(final ChannelTypeUID channelTypeUID, @Nullable final Locale locale) {
+        return generatedChannelTypes.get(channelTypeUID);
+    }
+
+    public StateChange checkStateChangeValid(SensiboSky sensiboSky, String property, Object value) {
+
```

Comment:
remove this blank line

Reply:
Fixed. Once the next sat plugin is released we can bump the spotless code formatter version and have these issues fixed automatically. https://github.com/openhab/static-code-analysis/pull/376